lang/ocaml/Portfile

```diff
@@ -49,35 +49,49 @@ if {${os.major} < 11} {
     # need to add strnlen patch as not in library
     patchfiles-append  patch-strnlen-socketaddr.diff
 
-    # compiler selection pulls in clang-3.4 on 10.6 and less
-    # and this compiler is needed for this port to run once installed
-    # so force this compiler and require it at runtime
-    compiler.whitelist     macports-clang-3.4
-    depends_run-append     port:clang-3.4
+	if {${build_arch} eq "i386" || ${build_arch} eq "x86_64"} {
+		# compiler selection pulls in clang-3.4 on 10.6 and less
+		# and this compiler is needed for this port to run once installed
+		# so force this compiler and require it at runtime
+		compiler.whitelist     macports-clang-3.4
+		depends_run-append     port:clang-3.4
+	}
 }
 
 # Note: the port maintainer has no access to an i386 host, and cannot
 # debug problems on one.
-if {${build_arch} eq "i386"} {
+if {${build_arch} eq "i386" || ${build_arch} eq "ppc"} {
     patchfiles-append  patch-configure-darwin32.diff
 }
 
+if {${build_arch} eq "ppc" || ${build_arch} eq "ppc64"} {
+    patchfiles-append  patch-configure-darwin-ppc.diff
+    compiler.blacklist-append *clang*
+    configure.args-append --enable-imprecise-c99-float-ops
+}
+
```

Comment:
Is the imprecise-c99-float-ops actually necessary?

Reply:
For Rosetta, necessary, otherwise the build fails. For other configurations doesn’t hurt, but not required.

Reply:
Why are we building under Rosetta? I'm also not convinced that it can't hurt if people have concerns about their numerics. The OCaml documentation specifically warns against this flag. It should only be turned on in rare circumstances if absolutely needed because of flaws in c99 support.